csharp_dotnetcore/60.ConversationHistory/Startup.cs

```diff
@@ -0,0 +1,94 @@
+                .AddJsonFile("appsettings.json")
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        /// <summary>
+        /// Gets the configuration that represents a set of key/value application configuration properties.
+        /// </summary>
+        /// <value>
+        /// The <see cref="IConfiguration"/> that represents a set of key/value application configuration properties.
+        /// </value>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> specifies the contract for a collection of service descriptors.</param>
+        /// <seealso cref="IStatePropertyAccessor{T}"/>
+        /// <seealso cref="https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/dependency-injection"/>
+        /// <seealso cref="https://docs.microsoft.com/en-us/azure/bot-service/bot-service-manage-channels?view=azure-bot-service-4.0"/>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            AzureBlobTranscriptStore transcriptStore = null;
+            services.AddBot<ConversationHistoryBot>(options =>
+            {
+                var botConfig = BotConfiguration.Load(@".\ConversationHistory.bot");
+                var service = botConfig.Services.FirstOrDefault(s => s.Type == "endpoint");
+                var endpointService = service as EndpointService;
+                if (endpointService == null)
+                {
+                    throw new InvalidOperationException("The .bot file does not contain an endpoint.");
+                }
+
+                options.CredentialProvider = new SimpleCredentialProvider(endpointService.AppId, endpointService.AppPassword);
+
+                // The Memory Storage used here is for local bot debugging only. When the bot
+                // is restarted, everything stored in memory will be gone.
+                IStorage dataStore = new MemoryStorage();
+
+                // Create Conversation State object.
+                // The Conversation State object is where we persist anything at the conversation-scope.
+                var conversationState = new ConversationState(dataStore);
+
+                options.State.Add(conversationState);
+
+                // Enable the conversation transcript middleware.
+                var connectionString = Configuration.GetValue<string>("ConnectionString");
+                transcriptStore = new AzureBlobTranscriptStore(connectionString, "transcriptstore");
+                var transcriptMiddleware = new TranscriptLoggerMiddleware(transcriptStore);
+                options.Middleware.Add(transcriptMiddleware);
+
+            });
+
+            services.AddSingleton(_ => { return new TranscriptStore(transcriptStore); });
+        }
```

Comment:
This method is odd due to the double naming of TranscriptStore / transcriptStore. Is there a better option?

Reply:
I've renamed transcriptStore to blobstore.

Reply:
After some consideration, the TranscriptStore seems superfluous 

---
In reply to: [218236540](https://github.com/Microsoft/BotBuilder-Samples/pull/391#discussion_r218236540) [](ancestors = 218236540)